base_layer/wallet_ffi/src/callback_handler.rs

```diff
@@ -385,6 +385,19 @@ where TBackend: TransactionBackend + 'static
     }
 
     async fn receive_transaction_cancellation(&mut self, tx_id: TxId) {
+        match self.db.get_any_transaction(tx_id).await {
+            Ok(_) => {
+                // Make sure the transaction is cancelled
+                self.db.cancel_pending_transaction(tx_id).await.unwrap_or_default();
```

Comment:
I disagree with this. This callback handler should not be contributing to changing state at all. It monitors the event streams and calls callbacks. If this is a bug then it really needs to be addressed at the point in the code that would produce these events. This is going to create a different behaviour in a mobile wallet using the FFI vs the console wallet.